shopinvader/services/partner_mixin.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019 Camptocamp (http://www.camptocamp.com).
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# pylint: disable=method-required-super, consider-merging-classes-inherited
+
+from odoo import _
+from odoo.addons.component.core import AbstractComponent
+
+
+class PartnerServiceMixin(AbstractComponent):
+    _name = "shopinvader.partner.service.mixin"
+
+    def _post_create(self, partner):
```

Comment:
@simahawk That makes a lot of code :smirk: Would-it be better to implement the notification system into an event handler component registered on "shopinvader.partner"?

Reply:
that's why is a mixin. To me it strictly belongs to partner-related services and having a separated component sounds overkilling IMO, but I'm open to suggestions :)

Reply:
IMO mixins are a false good idea.  With Odoo we are formatted to implement only inheritance-based extension solutions. This goes against some good practices of object/component-oriented programming.  It also prevents us from thinking in terms of separation of responsibilities (everything related to a model is implemented by the model) . I prefer to delegate some part of the logic to separate components in charge of a specific fonctional domain (ex salesman notification).

Reply:
@lmignon pls check my comment above.  I think this is strictly tied to the service itself since as you can see each service overrides and/or customizes some bits to change the logic (eg: check if the customer is enabled for that service or not or get the specific notification for that service).
So, I think we'll have to register 2 components (1 per each service) + a mixin to share the logic I don't see how you could make it more clean.
Otherwise you'll have to put service-specific logic into 1 component and discriminate w/ some var passed around and some IF blocks here and there.
See my point?

OTOH we could go w/ current implementation so that we have the feature and we can test it in real world. Then we can think better about how to refactor it if needed.

Reply:
I see your point but I'm not convinced at all. No time to dig deeper into this code so let's moving forward...

Reply:
@lmignon I've refactored out to component the validation part. It makes the mixin much lighter. I'll add a TODO for refactoring as component event machinery. Sounds good?

Reply:
ok